### PR TITLE
(874) Assess - Suitability assessment questions

### PR DESCRIPTION
**Pull request by richpjames**
# Context
This PR adds a Suitability Assessment questions page to the Assess stage of the journey

[Trello ticket](https://trello.com/c/GbIxeHTc/874-suitability-assessment-questions)

# Changes in this PR
- The prototype had incorrectly named the previous section 'Suitability Assessment' so first we clear that up 
- Then we add a section for 'Assess' to the Assess journey
- Then we add the Page Class
- Then we add the view. We try to keep as much creation of html/view elements as possible in helper functions as opposed to page classes. Here both the page class and the view need to share some information so rather than duplicate I pass it to the view through the utils file. We then use this information to iterate through and create the view.
It’s not perfect: ideally the suitabilityAssessmentSections would be in the utils file but that would createa cyclical dependency.

## Screenshots of UI changes

Assess -- allows me to assess an application (4)

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
    `production` in a backwards compatible way? (This includes our API,
    infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
    advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/form-pages/assess/assessApplication/suitablityAssessment/suitabilityAssessment.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import { Page } from '../../../utils/decorators'
+import { sentenceCase } from '../../../../utils/utils'
+
+import TasklistPage from '../../../tasklistPage'
+
+export const suitabilityAssessmentSections: Array<SuitabilityAssessmentSection> = [
+  {
+    id: 'riskFactors',
+    question: 'Does the application identify the risk factors that an Approved Premises (AP) placement can support?',
+    guidanceNotes: `<p>Consider:</p>
```
Comment (pezholio) [inline server/form-pages/assess/assessApplication/suitablityAssessment/suitabilityAssessment.ts:12]: I'm wondering if this HTML is better placed in Nunjucks partials, as there's a lot of it. So we could have them in 
`views/assessments/pages/partials/$ID` and then load them in that way. Then the sections could within the Page class instance itself like so:

```
export default class SuitabilityAssessment implements TasklistPage {
   ....

   sections = [
      {
          id: 'riskFactors',
          question: 'Does the application identify the risk factors that an Approved Premises (AP) placement can support?'
      },
      ...
   ]
}
```

And then used in the view like so:

```
{% for section in page.sections %}
        {{ formPageRadios({
                fieldName: section.id,
                fieldset: {
                    legend: {
                        text: section.question,
                        classes: "govuk-fieldset__legend--s"
                    }
                },
                items: [
                    {
                        value: "yes",
                        text: "Yes"
                    },
                    {
                        value: "no",
                        text: "No"
                    }
                ]
            }, fetchContext()) }}

        <details class="govuk-details" data-module="govuk-details">
            <summary class="govuk-details__summary">
                <span class="govuk-details__summary-text">
                          View guidance notes
                        </span>
            </summary>

            <div class="govuk-details__text">
                {% include './partials/' + section.id + '.njk' %}
            </div>
        </details>

        {{ formPageTextarea({
                    fieldName: section.id + "Comments",
                    label: {
                        text: "Additional comments",
                        classes: "govuk-label--s"
                    }
                }, fetchContext()) }}
{% endfor %}
```

That way we keep the page class neater without a ton of HTML, and we don't have to pass the sections variable to Nunjucks.

Whaddya reckon?

Comment (richpjames) [reply server/form-pages/assess/assessApplication/suitablityAssessment/suitabilityAssessment.ts:12]: Yeah seems smart, I'll give it a go

File: server/utils/assessmentUtils.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ import { embeddedSummaryListItem } from './checkYourAnswersUtils'
 import reviewSections from './reviewUtils'
 import Apply from '../form-pages/apply'
 import { kebabCase } from './utils'
+import { suitabilityAssessmentSections as suitabilityAssessmentQuestions } from '../form-pages/assess/assessApplication/suitablityAssessment/suitabilityAssessment'
```
Comment (pezholio) [inline server/utils/assessmentUtils.ts:16]: Could we not have this as a property on the form?

Comment (richpjames) [reply server/utils/assessmentUtils.ts:16]: Yeah that makes sense

File: server/form-pages/assess/assessApplication/suitablityAssessment/suitabilityAssessment.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    'moveOnPlan',
+    'moveOnPlanComments',
+  ],
+})
+export default class SuitabilityAssessment implements TasklistPage {
+  name = 'suitability-assessment'
+
+  title = 'Suitability assessment'
+
+  sections: Array<SuitabilityAssessmentSection> = [
```
Comment (pezholio) [inline server/form-pages/assess/assessApplication/suitablityAssessment/suitabilityAssessment.ts:31]: One more improvement (sorry!). Could sections just be a key/value record now? So:

```
sections = {
  riskFactors: 'Does the application identify the risk factors that an Approved Premises (AP) placement can support?',
...
}
```

**Review by pezholio — APPROVED** [review]
LGTM - thanks for bearing with me on this one! 👍

**Pull request merged**